core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -490,6 +490,15 @@ public SELF withEnv(String key, String value) {
      * {@inheritDoc}
      */
     @Override
+    public SELF withEnv(Map<String, String> env) {
```

Comment:
I would have preferred to do that but unfortunately then the `withEnv` method would return `Container` rather than `GenericContainer`, which is annoying. I've raised #151 because I've run into this type-casting problem before.